hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java

```diff
@@ -275,6 +280,11 @@ public QueryResult newQueryResult(int numberOfPartitions) {
         return new QueryResult(queryResultSizeLimiter.getNodeResultLimit(numberOfPartitions));
     }
 
+    @Override
+    public Predicate optimize(Predicate predicate, Indexes indexes) {
```

Comment:
I don't understand why exposing it is relevant to the number of calls to optimize. The underlying implementation is smelly because the queryengine doesnt deal with queries using index. 

Imho this should really be an implementation detail.